fleet/postgres/agents.go

```diff
@@ -30,6 +30,41 @@ type agentRepository struct {
 	logger *zap.Logger
 }
 
+func (r agentRepository) RetrieveMatchingAgents(ctx context.Context, tags types.Tags) (types.Metadata, error) {
+	t, tmq, err := getOrbTagsQuery(tags)
```

Comment:
To be clear, whenever we are matching an agent group to agents, it must match against _both_ `agents.agent_tags` and `agents.orb_tags` - is that what we are doing here and in the other calls? You will see that the query for the `agent_group_membership` does this, for example.